hedera-node/docs/design/services/token-service/airdrops/token-cancel-airdrop.md

```diff
@@ -0,0 +1,143 @@
+     * This transaction MUST be signed by _each_ account *sending* an airdrop to be canceled.
+     */
+    rpc cancelAirdrop (Transaction) returns (TransactionResponse);
+}
+```
+
+### Fees
+
+A sender may cancel pending transfers for a low nominal fee using the new `TokenCancelAirdrop` transaction. The fee should be proportional to the number of airdrops being canceled in the transaction.
+
+An update into the `feeSchedule` file would be needed to specify that.
+
+### Services updates
+
+- Update `TokenServiceDefinition` class to include the new RPC method definition for cancelling airdrops
+- Implement new `TokenCancelAirdropHandler` class which should be invoked when the gRPC server handles `TokenCancelAirdrop` transactions. The class should be responsible for:
+    - Pure checks: validation logic based only on the transaction body itself in order to verify if the transaction is valid one
+        - Verify that the pending airdrops list contains between 1 and 10 entries, inclusive
+        - Verify that the pending airdrops list does not have any duplicate entries
+    - Pre-handle:
+        - The transaction must be signed by the account referenced by a `sender_id` for each entry in the pending airdrops list
+    - Handle:
+        - Confirm that for the given pending airdrops ids in the transaction there are corresponding pending transfers existing in state
+        - Any additional validation depending on config or state i.e. semantics checks
+        - The business logic for cancelling pending airdrops
+            - Should boil down to clearing up the pending airdrops entries from the pending airdrops state
+    - Fees calculation
+- Additional considerations:
+    - All pending transfers sent by an account must be canceled before the account can be deleted
+    - If the sender’s account expires and cannot be renewed, then all pending transfers for that sender are canceled
+
+## Acceptance Tests
+
+All of the expected behaviour described below should be present only if the new `TokenCancelAirdrop` feature flag is enabled.
+
+- Given existing pending airdrop in state when valid `TokenCancelAirdrop` transaction containing entry for the same pending airdrop is performed then the `TokenCancelAirdrop` should succeed and the pending airdrop should be removed from state
+- Given successful  `TokenCancelAirdrop` when `TokenClaimAirdrop` for the same airdrop is performed then the `TokenClaimAirdrop` should fail
+- Given successful  `TokenCancelAirdrop` then any future rents for the canceled pending airdrop should not be charged from the sender
+- Given account with pending airdrops in state that cannot be deleted when successful `TokenCancelAirdrop` is performed for all of the account's pending airdrops then the account can be successfully deleted
+- Given account that expires and cannot be renewed when that account is used as sender in pending airdrops then all those pending airdrops should be canceled
```

Comment:
While this will be needed in the future, I do not think we can implement it right now and therefore it should not be an acceptance test.